ibtk/include/ibtk/JacobianCalculator.h

```diff
@@ -98,37 +195,98 @@ class Mapping : public JacobianCalculator
     /*!
      * Constructor.
      */
-    Mapping(const key_type quad_key);
+    NodalMapping(const key_type quad_key, const FEUpdateFlags update_flags);
+
+    /*!
+     * Recalculate relevant quantities for the provided element.
+     */
+    virtual void reinit(const libMesh::Elem* elem) override;
 
     /*!
      * Calculate the JxW values on the given element and return a reference to
      * the result.
+     *
+     * @deprecated This function only exists for compatibility with the older
+     * JacobianCalculator class.
      */
     const std::vector<double>& get_JxW(const libMesh::Elem* elem) override
     {
-        return get(elem).d_JxW;
+        reinit(elem);
+        return d_JxW;
+    }
+
+    virtual const std::vector<double>& getJxW() const override
```

Comment:
For these kinds of operations, would we be better off templating over the mapping class, to facilitate inlining, rather than using virtual functions?

Reply:
We should probably just make these not virtual but I'd prefer to leave it this way for now - the overhead is small enough that its not worth worrying about yet.